fix: prevent conflicts with old plugin

jasonbahl opened the pull request:
What does this implement/fix? Explain your changes.
---------------------------------------------------
This ensures  actions and filters are unique to this version of the plugin. This means that any custom code targeting hooks and filters of the previous WPGraphQL for ACF will not execute when this plugin is active and won't cause unintended side effects.

This also adjusts Class names to be unique to ensure any code from extending codebases don't accidentally all code from this version of the plugin. 

What currently open issues does this close or contribute?
------------------------------------------

closes #65
closes #66 

Other Notes
------------------------------------------

### Failing Schema Linter

Note, the Schema Linter test is failing because the previous release action did not upload the Schema Artifact to the release: https://github.com/wp-graphql/wpgraphql-acf/releases/tag/v2.0.0-beta.3.1.0

### Action / Filter comparison across versions

2.0.0-beta.4.0 converts to a `wpgraphql/acf/${hook_name}` convention. 

We believe this is a bit more readable and makes it more clear what the filter/hook is intended for. 

We also believe this will help with folks upgrading. If anyone had custom code that extended v0.6 or prior, it might be easy to miss that it's not working because the hook name is `graphql_acf_*` instead of `wpgraphql_acf_*` where it might be easily overlooked that the `wp` was added, thus causing much time banging heads against a desk and pulling hair out before realizing the change.

We believe it will be more clear if the user is looking at docs for the latest version of WPGraphQL and sees `wpgraphql/acf/*` they would know immediately that their old action with `graphql_acf_*` would not be firing, thus saving hair and desks.

| v0.6.3                                           | v2.0.0-beta.3.1.0                              | v2.0.0-beta.4.0                                  |
|--------------------------------------------------|------------------------------------------------|--------------------------------------------------|
| wpgraphql_acf_should_field_group_show_in_graphql | graphql_acf_should_field_group_show_in_graphql | wpgraphql/acf/should_field_group_show_in_graphql |
| graphql_acf_get_root_id                          | n/a                                            | wpgraphql/acf/pre_get_node_acf_id                |
| graphql_acf_field_value                          | graphql_acf_field_value                        | wpgraphql/acf/field_value                        |
| wpgraphql_acf_supported_fields                   | wpgraphql_acf_supported_field_types            | wpgraphql/acf/supported_field_types              |
| wpgraphql_acf_register_graphql_field             | n/a                                            | n/a                                              |
| graphql_acf_post_object_source                   | n/a                                            | n/a                                              |
| n/a                                              | n/a                                            | wpgraphql/acf/get_graphql_field_config           |
| n/a                                              | graphql_acf_is_acfe_active                     | wpgraphql/acf/is_acfe_active                     |
| n/a                                              | graphql_acf_field_type_default_admin_settings  | wpgraphql/acf/field_type_default_admin_settings  |
| n/a                                              | graphql_acf_field_type_admin_settings          | wpgraphql/acf/field_type_admin_settings          |
| n/a                                              | graphql_acf_excluded_admin_field_settings      | wpgraphql/acf/excluded_admin_field_settings      |
| n/a                                              | graphql_acf_pre_resolve_acf_field              | wpgraphql/acf/pre_resolve_acf_field              |
| n/a                                              | graphql_acf_get_registered_field_types         | wpgraphql/acf/get_registered_field_types         |
| graphql_acf_init                                 | graphql_acf_init                               | wpgraphql/acf/init                               |
| graphql_acf_match_location_rule                  | graphql_acf_match_location_rule                | wpgraphql/acf/match_location_rule                |
| n/a                                              | graphql_acf_register_field_types               | wpgraphql/acf/register_field_types               |
| n/a                                              | graphql_acf_registry_init                      | wpgraphql/acf/registry_init                      |

jasonbahl committed:
- add slevomat/coding-standard as a dependency
- update phpcs.xml.dist rules to better align with WPGraphQL core

jasonbahl committed:
- delete un-used phpcs.ruleset.xml

jasonbahl committed:
- add test to ensure old classes do not exist. This should prevent us from accidentally adding classes that match the old version of the plugin. We don't want old functions to be available in the new plugin, intentionally.

jasonbahl committed:
- update "tested up to" flag to 6.2
- update testing-integration workflow to test against WP 6.2

jasonbahl committed:
- update testing matrix

jasonbahl committed:
- fix typo

jasonbahl committed:
- revert coverage workflow to use WP 6.1

markkelnar commented:
I see the action '[graphql_acf_match_location_rule](https://github.com/wp-graphql/wpgraphql-acf/pull/67/files#diff-3833bf731abfe852e542919f7ceaf808c2712d82d9a8566487757cad7d38009aR345)'. Does that need to change?

markkelnar commented:
Is it intentional not to change the name of the actions that are triggered in Field Type Registry? graphql_acf_registry_init and  graphql_acf_register_field_types.

jasonbahl committed:
- update action names

jasonbahl commented:
> Is it intentional not to change the name of the actions that are triggered in Field Type Registry? graphql_acf_registry_init and graphql_acf_register_field_types.

@markkelnar looks like I updated the filters but not the actions 🤦🏻‍♂️. Good catch. I updated the actions and the table documenting the differences.

jasonbahl committed:
- update action name for AcfExtended support

jasonbahl committed:
- update action

jasonbahl committed:
- update action `register_graphql_acf_field_type` hooks into

jasonbahl committed:
- update hook/filter names to use a "namespace" style of naming with forward slashes

jasonbahl committed:
- update phpcs.xml.dist to allow for action/filter names to include forward slashes without causing a linting error
